### ISSUE TITLE
if_running unexpected behaviour + requires extra white-space

### ISSUE THREAD

**Issue by Nooby4Ever** [brndnmtthws/conky#121]
On Linux.
1) You're forced to give the full verbose name of the process you want to check, i think this is not (always) desired; or at least we should be given the option not to.

2) If there is no white-space behind the statement eg <code>${if_running firefox}</code> it will evaluate to false while <code>${if_running firefox }</code> will evaluate to true.
***
Minimal example (run conky in the console, with firefox running):
```
conky.config = {
	out_to_x = false,
	out_to_console = true,
};


conky.text = [[
${if_running conky }Running True${else}Running False${endif}
${if_running conky -c /home/nooby4ever/Desktop/test }Running True2${else}Running False2${endif}
${if_running firefox }Running True3${else}Running False3${endif}
${if_running firefox}Running True4${else}Running False4${endif}
]];
```
Output:
```
Running False
Running True2
Running True3
Running False4
```
While this may be more accurate, eg: when running multiple conkies you can check whether or not a specific instance is running instead of just "an instance" is running but for some applications this accuracy is not required/desired.
At least it needs to be specified in the documentation and i would suggest giving the user the option (short or verbose name); unfortunately my C/C++ skills are not enough to code it by myself.

In essence it's rather similar to: https://github.com/brndnmtthws/conky/issues/113

**Comment by marcpayne** [comment]
@Nooby4Ever I'll take a look at how `if_running` is handled as time permits. Checking the git log might give some insight into the rationale behind the new behavior. I try to give the benefit of the doubt before hacking away. However, based on your report here, there is definitely a bug that needs to be fixed (namely the requirement for a trailing space to make `if_running` evaluate to true).

**Comment by marcpayne** [comment]
OK, I finally got some time to look at this problem. The trailing space issue is really interesting to me because some `/proc/<pid>/cmdline` entries (e.g. `chromium --type=gpu-broker`) are padded with a *lot* of extra null characters for some reason, which are all replaced with spaces by conky's code in `linux.cc`. This part shouldn't be too hard to fix, hopefully.

BTW, nice job spotting this issue. Your report made my search much simpler. :)

To fix the issue about having to match against full command lines, I need to research some of the C string functions. I'm not that familiar with C yet.

This comment is partly a "note to self," and partly a note to C gurus who might want to tackle the issues themselves.

**Comment by Nooby4Ever** [comment]
Thank you, i always try to pin-point the bugs to my best knowledge so people who have the knowledge to fix the bugs themselves have an easier life. At least that's my intention though :)

**Comment by marcpayne** [comment]
I removed the branch containing the commits referenced above since there was a problem I hadn't considered. Not good git practice, but no harm done.

**Comment by marcpayne** [comment]
@Nooby4Ever Can you please test my latest batch of commits to see if they fix this issue for you? Everything is in the `basename` branch in my repo at https://github.com/marcpayne/conky.git. You can either clone my repo and checkout `basename`, or do something like the following in your local repo:

```
git remote add marcpayne https://github.com/marcpayne/conky.git
git fetch marcpayne
git checkout -b mybasename marcpayne/basename
```

I would greatly appreciate it! This branch has several changes and I think it's best to get some verification before I submit a PR.

**Comment by Nooby4Ever** [comment]
TL;DR: I can confirm it's working as intended.

***
Tested using (text section only);
```
conky.text = [[
Running ${if_running conky} True${else} False${endif}
Running ${if_running conky -c /home/nooby4ever/Desktop/test} True2${else} False2${endif} # Running

Running ${if_running conky -c /home/nooby4ever/Desktop/test2} True3${else} False3${endif} # Not running

Running ${if_running firefox} True4${else} False4${endif}
Running ${if_running firefox } True5${else} False5${endif}

${top name 1}
${top name 2}
${top name 3}
]];`
```
Results in (top results removed for convenience), with <code>top_name_verbose = true</code>:
```
Running  True
Running  True2 
Running  False3 
Running  True4
Running  False5
```
with <code>top_name_verbose = false</code>:
```
Running  True
Running  True2 
Running  False3 
Running  True4
Running  False5
```

Note: now the extra space at the end (test 5) will evaluate into false, this should be clearly documented (tailing spaces are not allowed).

I can also confirm processes with white-spaces in their name are correctly displayed when <code>top_name_verbose = false</code> is set. This will indeed behave as Conky 1.9 did. 

Although this is desired i must note Conky 1.9 had some unexpected behaviour in that regard. For example banshee (music player) will be shown as "Main Thread" instead of "banshee". As this was also happening in Conky 1.9 it is not a comment on your commit but rather on Conky's "old" behaviour. 
Running: <code>ps aux </code> gives the command as Conky 1.10 gives (or with your patch with the <code>top_name_verbose = true</code>) and <code>ps -A</code> give the command as Conky 1.9 gives (or with your patch with the <code>top_name_verbose =false</code>). This might also be banshee's fault since i have had no other program with this behaviour (and don't know the technical difference between "aux" and "-A" as to why there is a difference).
But since no one has complained about this in the past, and it's not a real issue for me i don't think you should really bother; but i thought it was worth noting. (This also means for banshee i have to use  <code>Running ${if_running Main Thread} True6${else} False6${endif}</code> or the full command)

Reminder: don't forget to update the nano and vim syntax highlight with <code>top_name_verbose</code>

**Comment by marcpayne** [comment]
Thank you for your tests!

>Note: now the extra space at the end (test 5) will evaluate into false, this should be clearly documented (tailing spaces are not allowed).

Correct, and I'll take note of the need for documentation.

>Reminder: don't forget to update the nano and vim syntax highlight with top_name_verbose

Thanks for reminding me. This has been on my radar for a while. Just haven't gotten around to it.

Banshee sounds like an odd beast. This "Main Thread" must be coming from `/proc/<pid>/stat`, which is disappointing to me since `stat` seemed like a 100% reliable source! Oh well. :)

**Comment by marcpayne** [comment]
This issue has been resolved since 1.10.1. Feel free to close it if you agree.

I have the vim and nano syntax highlight changes sitting in my local repo and will submit a PR after the Xinerama fix gets accepted. This is to avoid a merge conflict.

**Comment by Nooby4Ever** [comment]
I did not follow the PRs after #140 but i assume you've updated the syntax highlighting. Therefore i will close this issue.
